application/src/userlistmodel.cpp

```diff
@@ -0,0 +1,72 @@
+#include "userlistmodel.h"
+#include "databaseengine.h"
+
+UserListModel::UserListModel(QObject *parent) : QAbstractListModel(parent)
+{
+    connectDatabase();
+}
+
+void UserListModel::setUserlist(const QList<User> &userList)
+{
+    beginResetModel();
+    m_userList = userList;
+    emit countChanged();
+    endResetModel();
+}
+
+void UserListModel::addUserlist(const QList<User> &userList)
+{
+    beginInsertRows(QModelIndex(),m_userList.size(),m_userList.size()+userList.size()-1);
```

Comment:
```suggestion
    beginInsertRows(QModelIndex(), m_userList.size(), m_userList.size()+userList.size()-1);
```
code-style
Однако я не понимаю, в чем суть и необходимость этого метода?

Reply:
Если с сервера пришло новые контакты,чтобы всю модель не обновлять просто добавляются снизу

Reply:
а такое бывает? сервер умеет это?